File: src/BaseDeDatos.java
import java.util.ArrayList;

public class BaseDeDatos {

    companion object {

        val datos: ArrayList<Int> = ArrayList()

        fun recuperarPorIndice(indice: Int) : Int? {
            for (indiceArreglo in datos.indice){
                if (indiceArreglo == indice) {
                    return datos[indiceArreglo]
                }
            }
            return null
        }
    }
}

File: src/BaseDeDatos.kt
import java.util.ArrayList;

public class BaseDeDatos {

    companion object {

        val datos: ArrayList<Usuario> = ArrayList()

        fun recuperarPorNombreUsuario(nombreUsuario: String) : Usuario? {
            for (usuario in datos){
                if (usuario.nNombreUsuario.equals(nombreUsuario)) {
                    return usuario
                }
            }
            return null
        }
    }
}
